ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -48,10 +48,17 @@ end = struct
   let accept_cached = ref None
 
   let _restart_no_cache ~__context ~accept =
+    let client_auth_name =
```

Comment:
Do we have a chicken/egg problem on secondary hosts here? I believe this code will query the db just before starting stunnel for the first time on a _secondary_ host, but if we have db access that implies stunnel is already running ..?

I wonder if we even need to add this extra bit of stunnel config on secondary hosts in the first place (since all connections from the outside are coming in via the primary host)

Reply:
Keep in mind that there are separate stunnel processes for outgoing and incoming connections. This config is for the incoming one only, and I don't think that this problem exists.

Having said that, I only wanted to expose this functionality on the the primary host, and I'll need to add some code to do that.

Reply:
Updated.

Reply:
yup, you are right :)